qpid/cpp/src/qpid/broker/amqp/Outgoing.cpp

```diff
@@ -108,6 +109,19 @@ void OutgoingFromQueue::write(const char* data, size_t size)
     pn_link_send(link, data, size);
 }
 
+void OutgoingFromQueue::maybeMergeMessageAnnotations(const Record &r)
```

Comment:
Minor, subjective point on the method name: 'maybe' sounds vague, perhaps mergeAnnotationsIfRequired? or checkAndMergeAnnotations?

Change looks good though, thanks again!